x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJob.java

```diff
@@ -359,6 +359,14 @@ public static ConnectorSyncJob fromXContent(XContentParser parser) throws IOExce
         return PARSER.parse(parser, null);
     }
 
+    public static Connector syncJobConnectorFromXContentBytes(BytesReference source, XContentType xContentType) {
```

Comment:
Using this method inside `ConnectorSyncJobIndexService` will be a separate PR as this needs some more thinking how to correctly handle the `id` of the connector (when, where and how to set it).

Reply:
Sure, if you want to pass connector_id to the connector parser from context you can you can change ` SYNC_JOB_CONNECTOR_PARSER.parse(parser, null);` to `SYNC_JOB_CONNECTOR_PARSER.parse(parser, docId);` and set the Class `id` param in `ConstructingObjectParser`

Reply:
I actually went down this path already and encountered one minor issue/inconvenience, but nothing blocking. We can discuss that in a 1on1 👍